### PR TITLE
Starting to use the client wrapper in hbase 2 data.

### PR DESCRIPTION
**Pull request by sduskis**

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/core/IBigtableDataClient.java [hunk 1]
```diff
@@ -153,4 +155,12 @@
...
   //TODO(rahulkql): Once it is adapted to v2.models, change the return type to ApiFuture.
   ListenableFuture<List<FlatRow>> readFlatRowsAsync(Query request);
+
+  /**
+   * Read {@link FlatRow} asynchronously, and pass them to a stream observer to be processed.
+   *
+   * @param request  a {@link Query} object.
+   * @param observer a {@link StreamObserver} object.
+   */
+  void readFlatRowsAsync(Query request, StreamObserver<FlatRow> observer);
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/core/IBigtableDataClient.java]: Whats the future plan of FlatRow vs model Row?

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncTable.java [hunk 2]
```diff
@@ -95,7 +98,8 @@ public BigtableAsyncTable(BigtableAsyncConnection asyncConnection,
       HBaseRequestAdapter hbaseAdapter) {
     this.asyncConnection = asyncConnection;
     BigtableSession session = asyncConnection.getSession();
-    this.client = new BigtableDataClient(session.getDataClient());
+    this.client = new BigtableDataClient(session.getDataClient(), session.getClientWrapper());
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncTable.java]: Maybe add a comment that the unwrapped data client is going away, once the wrapper is done

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/core/IBigtableDataClient.java (hunk 1 above)
Comment (sduskis) [reply bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/core/IBigtableDataClient.java]: I have no idea.  Let's discuss that offline.  For now, I'd rather continue as is.

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncTable.java (hunk 2 above)
Comment (sduskis) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncTable.java]: We have this pattern all over the place.  I'd rather not comment here only.

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/core/IBigtableDataClient.java (hunk 1 above)
Comment (sduskis) [reply bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/core/IBigtableDataClient.java]: I created #2077 to track this.

**Pull request merged**